lib/src/main/java/com/nextcloud/android/sso/SingleSignOn.java

```diff
@@ -0,0 +1,52 @@
+package com.nextcloud.android.sso;
+
+import com.nextcloud.android.sso.model.FilesAppType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+public class SingleSignOn {
```

Comment:
I don't like the name as it is kind of too generic. Maybe `FilesAppTypeRegistry` or anything else more verbose would be helpful?